src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -986,22 +779,440 @@ public void AnalyzeCommand_LoadConfigurationFile(string configValue, bool defaul
+
+            Uri updatedUri = new Uri(@"c:\updated\directory\newFileName.txt", UriKind.RelativeOrAbsolute);
+            Notification expectedNotification = BuildTestNotification(updatedUri);
+
+            AnalyzeCommandBase<TestAnalysisContext, AnalyzeOptionsBase>
+                .UpdateLocationsAndMessageWithCurrentUri(actualNotification.Locations, actualNotification.Message, updatedUri);
+
+            actualNotification.Should().BeEquivalentTo(expectedNotification);
+        }
+
+        private Notification BuildTestNotification(Uri uri)
+        {
+            string filePath = uri.OriginalString;
+            string fileName = Path.GetFileName(uri.OriginalString);
+            return new Notification
+            {
+                Locations = new List<Location>
+                {
+                    new Location
+                    {
+                        PhysicalLocation = new PhysicalLocation
+                        {
+                            ArtifactLocation = new ArtifactLocation
+                            {
+                                Uri = uri
+                            }
+                        }
+                    }
+                },
+                Message = new Message
+                {
+                    Arguments = new List<string> { filePath, fileName },
+                    Text = string.Format(@"Found an issue in {0} (full path is {1}", filePath, fileName)
+                }
+            };
+        }
+
+        [Fact]
+        public void AnalyzeCommandBase_GetFileNameFromUriWorks()
+        {
+            var sb = new StringBuilder();
+
+            var testCases = new Tuple<string, string>[]
+            {
+                new Tuple<string, string>(null, null),
+                new Tuple<string, string>(@"file.txt", "file.txt"),
+                new Tuple<string, string>(@".\file.txt", "file.txt"),
+                new Tuple<string, string>(@"c:\directory\file.txt", "file.txt"),
+                new Tuple<string, string>(@"\\computer\computer\file.txt", "file.txt"),
+                new Tuple<string, string>(@"file://directory/file.txt", "file.txt"),
+                new Tuple<string, string>(@"/file.txt", "file.txt"),
+                new Tuple<string, string>(@"directory/file.txt", "file.txt"),
+            };
+
+            foreach (Tuple<string, string> testCase in testCases)
+            {
+                Uri uri = testCase.Item1 != null ? new Uri(testCase.Item1, UriKind.RelativeOrAbsolute) : null;
+                string expectedFileName = testCase.Item2;
+
+                string actualFileName = AnalyzeCommandBase<TestAnalysisContext, AnalyzeOptionsBase>.GetFileNameFromUri(uri);
+
+                if (!object.Equals(actualFileName, expectedFileName))
+                {
+                    sb.AppendLine(string.Format("Bad file name returned for uri '{0}'. Expected '{1}' but saw '{2}'.", uri, expectedFileName, actualFileName));
```

Comment:
>Bad [](start = 49, length = 3)

"Bad" => "Incorrect"

Reply:
But that file name is bad. It's a very bad name and it doesn't get a treat tonight.

---
In reply to: [322903221](https://github.com/microsoft/sarif-sdk/pull/1670#discussion_r322903221) [](ancestors = 322903221)